chrome/content/zotero/xpcom/itemPaneManager.js

```diff
@@ -0,0 +1,340 @@
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+/**
+ * @typedef SectionIcon
+ * @type {object}
+ * @property {string} icon - Icon URI
+ * @property {string} [darkIcon] - Icon URI in dark mode. If not set, use `icon`
+ * @typedef SectionL10n
+ * @type {object}
+ * @property {string} l10nID - data-l10n-id for localization
+ * @property {string} [l10nArgs] - data-l10n-args for localization
+ * @typedef SectionButton
+ * @type {object}
+ * @property {string} type - Button type, must be valid DOMString and without ","
+ * @property {(props: SectionEventHookArgs) => void} onClick - Button click callback
+ * @typedef SectionBasicHookArgs
+ * @type {object}
+ * @property {string} paneID - Registered pane id
+ * @property {Document} doc - Document of section
+ * @property {HTMLDivElement} body - Section body
+ * @typedef SectionUIHookArgs
+ * @type {object}
+ * @property {Zotero.Item} item - Current item
+ * @property {string} tabType - Current tab type
+ * @property {boolean} editable - Whether the section is in edit mode
+ * @property {(l10nArgs: string) => void} setL10nArgs - Set l10n args for section header
+ * @property {(l10nArgs: string) => void} setEnabled - Set pane enabled state
+ * @property {(summary: string) => void} setSectionSummary - Set pane section summary
+ * @property {(buttonType: string, options: {disabled?: boolean; hidden?: boolean}) => void} setSectionButtonStatus - Set pane section button status
+ * @typedef SectionHookArgs
+ * @type {SectionBasicHookArgs & SectionUIHookArgs}
+ * @typedef  {SectionHookArgs & { refresh: () => Promise<void> }} SectionInitHookArgs
+ * A `refresh` is exposed to plugins to allows plugins to refresh the section when necessary,
+ * e.g. item modify notifier callback. Note that calling `refresh` during initialization
+ * have no effect.
+ * @typedef  {SectionHookArgs & { newData: Zotero.Item }} SectionDataChangeHookArgs
+ * @typedef  {SectionHookArgs & { event: Event }} SectionEventHookArgs
+ * @typedef ItemDetailsSectionOptions
+ * @type {object}
+ * @property {string} paneID - Unique pane ID
+ * @property {string} pluginID - Set plugin ID to auto remove section when plugin is disabled/removed
+ * @property {SectionL10n & SectionIcon} head - Header options. Icon should be 16*16 and `label` need to be localized
+ * @property {SectionL10n & SectionIcon} sidenav - Sidenav options. Icon should be 20*20 and `tooltiptext` need to be localized
+ * @property {string} [fragment] - Pane fragment as string
```

Comment:
Can you clarify what this means?

Reply:
In case plugin developers don't want to manually `createElement` everything, they can pass a string of HTML elements. (or, do you think it's better to be a XHTML file URL?)

It's like what we allow plugins to do in the pref pane - to write a `.xhtml` and load the content from it.

Reply:
Maybe @AbeJellinek has an opinion on this. If it were a file URL, I guess we'd have to cache it ourselves so it wasn't loaded every time, unlike in preferences where loading it from the file every time is fine.

Should clarify the variable name and comment, though, and we need to define whether this is HTML or XUL by default. We're calling this the "body" elsewhere. Should this be `bodyHTML`?

Reply:
> If it were a file URL, I guess we'd have to cache it ourselves so it wasn't loaded every time

Caching would introduce a lot of problems. Like when the plugin got updated, or reloaded during plugin development (where the file is changed but the plugin version is not). I would rather not face that.

`bodyHTML` instead of `fragment`? Sounds good. I think default to XUL?

Reply:
Yeah, `body` defaulting to XUL sounds good.

Reply:
done